heudiconv/_tmpdirs.py

```diff
@@ -0,0 +1,30 @@
+import logging
+import tempfile
+
+
+class TempDirs:
+    """A helper to centralize handling and cleanup of dirs"""
+
+    dirs = set()
+    lgr = logging.getLogger('heudiconv.tempdirs')
```

Comment:
all `lgr.` invocations were removed.  I think they should be brought back at debug level

Reply:
but it seems that their use was completely replaced with `tempfile.` constructs, then TempDirs isn't used at all, or what am I missing?

Although I am all for mitigating lingering temp dirs, and like that we would no longer pass them in functions (not sure why that was done originally anyways), I am not sure if switching to tempfile constructs wouldn't have detrimental DX effects:
- Absent easy to trigger logging to discover details on what is going on
- Absent centralization making it impossible to disable cleanup e.g. for the purpose of debugging

so eventually the centralized control might come back but would be improved to also provide context-manager style of invocation support .  So why not to do that right here and make it possible to do

```python
with tmpdirs(prefix='dcm2niix') as tmpdir:
```
which would take care about logging and cleaning it up right away (unless centrally disabled)?

Reply:
In my ideal world, we would purge the hacky `TempDirs` (which is just a remnant of the old single script days) and wrap all necessary code with `tempfile.TemporaryDirectory` context managers. Then but doing so requires restructuring / rewriting some of the logic (it all stems from `get_extracted_dicoms`), and I am not up for that task.

> Absent centralization making it impossible to disable cleanup e.g. for the purpose of debugging

Are we doing this now? This feels like it should be addressed in a separate PR.